src/fs.rs

```diff
@@ -151,6 +155,141 @@ pub fn create_workspace_archive(working_root_dir: &str, execution_id: &str) -> R
     }
 }
 
+pub fn create_yaml_backup_file<P>(
+    working_root_dir: P,
+    chart_name: String,
+    resource_name: Option<String>,
+    content: String,
+) -> Result<String, std::io::Error>
+where
+    P: AsRef<Path>,
+{
+    match resource_name.is_some() {
+        true => info!(
+            "creating yaml backup file for {}/{}",
+            chart_name,
+            resource_name.as_ref().unwrap()
+        ),
+        false => info!("creating yaml backup file for {}", chart_name),
+    };
+
+    let dir = working_root_dir.as_ref().join("backups");
+
+    create_dir_all(&dir)?;
+
+    let root_path = dir
+        .to_str()
+        .map(|e| e.to_string())
+        .ok_or_else(|| Error::from(ErrorKind::NotFound));
+
+    let string_path = match resource_name.is_some() {
+        true => format!(
+            "{}/{}-{}-q-backup.yaml",
+            root_path?,
+            chart_name,
+            resource_name.as_ref().unwrap()
+        ),
+        false => format!("{}/{}.yaml", root_path?, chart_name),
+    };
+    let str_path = string_path.as_str();
+    let path = Path::new(str_path);
+
+    let mut file = match File::create(&path) {
+        Err(e) => return Err(e),
+        Ok(file) => file,
+    };
+
+    match file.write(content.as_bytes()) {
+        Err(e) => Err(e),
+        Ok(_) => {
+            file.flush()?;
+            Ok(path
+                .to_str()
+                .map(|e| e.to_string())
+                .ok_or_else(|| Error::from(ErrorKind::NotFound))?)
+        }
+    }
+}
+
+pub fn remove_lines_starting_with(path: String, starter: &str) -> Result<String, std::io::Error> {
```

Comment:
OPT: Maybe return a CommandError here, so this function takes care of sending back safe and unsafe message properly.